### PR TITLE
Search pack by ID

### PR DESCRIPTION
**Pull request by ShahafBenYakir**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/28507

## Description
Changed the API path used in the search content module to use a path based on pack ID and not pack name. Needed to adjust unit tests accordingly.

### CODE REVIEW

**Comment by ShahafBenYakir** [comment]
It is the same PR exactly except for these two commits: 
https://github.com/demisto/content/pull/9530/commits/6a5615e2347750528cd80453ad252c51248afdad
https://github.com/demisto/content/pull/9530/commits/6908cb08ef564c1ef96a25950e198bebae363870

File: Tests/Marketplace/search_and_install_packs.py [hunk 1]
```diff
@@ -137,35 +126,41 @@ def search_pack(client, prints_manager, pack_display_name, thread_index, lock):
...
+
+                pack_data = {
+                    'id': result_object.get('id'),
+                    'version': result_object.get('currentVersion')
+                }
                 return pack_data
 
             else:
-                print_msg = 'Did not find pack {} in bucket.\n'.format(pack_display_name)
+                print_msg = 'Did not find pack: {} by its ID: {} in bucket.\n'.format(pack_display_name, pack_id)
```
Comment (yaakovi) [inline Tests/Marketplace/search_and_install_packs.py:149]:
```suggestion
                print_msg = 'Did not find pack "{}" by its ID "{}" in bucket.\n'.format(pack_display_name, pack_id)
```

File: Tests/Marketplace/search_and_install_packs.py [hunk 2]
```diff
@@ -137,35 +126,41 @@ def search_pack(client, prints_manager, pack_display_name, thread_index, lock):
...
 
         if 200 <= status_code < 300:
             result_object = ast.literal_eval(response_data)
-            search_results = result_object.get('packs', [])
-            pack_data = get_pack_data_from_results(search_results, pack_display_name)
-            if pack_data:
-                print_msg = 'Found pack {} in bucket!\n'.format(pack_display_name)
+
+            if result_object and result_object.get('currentVersion'):
+                print_msg = 'Found pack: {} by its ID: {} in bucket!\n'.format(pack_display_name, pack_id)
```
Comment (yaakovi) [inline Tests/Marketplace/search_and_install_packs.py:138]:
```suggestion
                print_msg = 'Found pack "{}" by its ID "{}" in bucket!\n'.format(pack_display_name, pack_id)
```

File: Tests/Marketplace/search_and_install_packs.py [hunk 3]
```diff
@@ -137,35 +126,41 @@ def search_pack(client, prints_manager, pack_display_name, thread_index, lock):
...
                 prints_manager.add_print_job(print_msg, print_color, thread_index, LOG_COLORS.GREEN)
                 prints_manager.execute_thread_prints(thread_index)
+
+                pack_data = {
+                    'id': result_object.get('id'),
+                    'version': result_object.get('currentVersion')
```
Comment (yaakovi) [inline Tests/Marketplace/search_and_install_packs.py:144]:
```suggestion
                    'version': result_object.get('currentVersion'),
```

File: Tests/Marketplace/search_and_install_packs.py [hunk 4]
```diff
@@ -137,35 +126,41 @@ def search_pack(client, prints_manager, pack_display_name, thread_index, lock):
     try:
         # make the search request
         response_data, status_code, _ = demisto_client.generic_request_func(client,
-                                                                            path='/contentpacks/marketplace/search',
-                                                                            method='POST',
-                                                                            body={"packsQuery": pack_display_name},
+                                                                            path=f'/contentpacks/marketplace/{pack_id}',
```
Comment (yaakovi) [inline Tests/Marketplace/search_and_install_packs.py:129]: nice workaround

**Review by yaakovi — COMMENTED** [review]
looks ok, add a link to a build where the pack name was modified.

File: Tests/Marketplace/search_and_install_packs.py (hunk 3 above)
Comment (ShahafBenYakir) [reply Tests/Marketplace/search_and_install_packs.py:144]: What? Why?

Comment (Itay4) [reply Tests/Marketplace/search_and_install_packs.py:144]: because it is yaakovi

Comment (ShahafBenYakir) [reply Tests/Marketplace/search_and_install_packs.py:144]: :trollface:

Comment (Itay4) [reply Tests/Marketplace/search_and_install_packs.py:144]: https://www.python.org/dev/peps/pep-0008/#when-to-use-trailing-commas

File: Tests/Marketplace/search_and_install_packs.py [hunk 5]
```diff
@@ -121,13 +109,14 @@ def get_pack_dependencies(client, prints_manager, pack_data, thread_index, lock)
         lock.release()
 
 
-def search_pack(client, prints_manager, pack_display_name, thread_index, lock):
+def search_pack(client, prints_manager, pack_display_name, pack_id, thread_index, lock):
```
Comment (Itay4) [inline Tests/Marketplace/search_and_install_packs.py:112]: add type hints

File: Tests/Marketplace/search_and_install_packs.py [hunk 6]
```diff
@@ -137,35 +126,41 @@ def search_pack(client, prints_manager, pack_display_name, thread_index, lock):
...
+
+                pack_data = {
+                    'id': result_object.get('id'),
+                    'version': result_object.get('currentVersion')
+                }
                 return pack_data
 
             else:
-                print_msg = 'Did not find pack {} in bucket.\n'.format(pack_display_name)
+                print_msg = 'Did not find pack "{}" by its ID "{}" in bucket.\n'.format(pack_display_name, pack_id)
```
Comment (Itay4) [inline Tests/Marketplace/search_and_install_packs.py:149]: use f-strings son

File: Tests/Marketplace/search_and_install_packs.py [hunk 7]
```diff
@@ -137,35 +126,41 @@ def search_pack(client, prints_manager, pack_display_name, thread_index, lock):
...
-                                                                                       status_code, msg)
+            err_msg = 'Search request for pack "{}" with ID "{}", failed with status code {}\n{}'.format(
+                pack_display_name,
+                pack_id, status_code,
+                msg)
             raise Exception(err_msg)
     except Exception as e:
-        err_msg = 'The request to search pack {} has failed. Reason:\n{}'.format(pack_display_name, str(e))
+        err_msg = 'Search request for pack "{}" with ID "{}", failed. Reason:\n{}'.format(pack_display_name, pack_id,
+                                                                                        str(e))
```
Comment (Itay4) [inline Tests/Marketplace/search_and_install_packs.py:163]: over indent here

File: Tests/Marketplace/search_and_install_packs.py (hunk 6 above)
Comment (ShahafBenYakir) [reply Tests/Marketplace/search_and_install_packs.py:149]: We will have some like this and some like this... I don't think it's better... it is more confusing.

File: Tests/Marketplace/Tests/search_and_install_packs_test.py [hunk 8]
```diff
@@ -114,10 +95,19 @@ def __init__(self):
...
+
+    def release(self):
+        return None
+
+
 def test_search_and_install_packs_and_their_dependencies(mocker):
     """
     Given
-    - Valid and invalid integrations paths.
+    - Valid packs.
```
Comment (Itay4) [inline Tests/Marketplace/Tests/search_and_install_packs_test.py:109]: what..

File: Tests/Marketplace/Tests/search_and_install_packs_test.py [hunk 9]
```diff
@@ -133,6 +123,7 @@ def test_search_and_install_packs_and_their_dependencies(mocker):
 
     client = MockClient()
 
+    mocker.patch.object(script, 'install_packs')
```
Comment (Itay4) [inline Tests/Marketplace/Tests/search_and_install_packs_test.py:126]: how is this related to your changes?

File: Tests/Marketplace/search_and_install_packs.py (hunk 6 above)
Comment (Itay4) [reply Tests/Marketplace/search_and_install_packs.py:149]: you already have it mixed, you modified the code yourself to be like that

File: Tests/Marketplace/Tests/search_and_install_packs_test.py (hunk 9 above)
Comment (ShahafBenYakir) [reply Tests/Marketplace/Tests/search_and_install_packs_test.py:126]: This test was throwing errors in the console for no reason. It should've been done when this was written.

File: Tests/Marketplace/Tests/search_and_install_packs_test.py (hunk 8 above)
Comment (ShahafBenYakir) [reply Tests/Marketplace/Tests/search_and_install_packs_test.py]: Two cases. first is a given valid pack ID second is given an invalid pack ID

Comment (ShahafBenYakir) [reply Tests/Marketplace/Tests/search_and_install_packs_test.py]: Added the word ID

**Comment by Itay4** [comment]
do we have a successful build ?

File: Tests/Marketplace/search_and_install_packs.py (hunk 6 above)
Comment (ShahafBenYakir) [reply Tests/Marketplace/search_and_install_packs.py:152]: Hhhhh no I did not but yalla

Comment (ShahafBenYakir) [reply Tests/Marketplace/search_and_install_packs.py]: Yalla changed

**Comment by ShahafBenYakir** [comment]
@Itay4 If you stop commenting then we can wait and see the 6.0 build here 👇

**Comment by yaakovi** [comment]
> @Itay4 If you stop commenting then we can wait and see the 6.0 build here 👇

failed

**Comment by ShahafBenYakir** [comment]
@yaakovi @Itay4 https://app.circleci.com/pipelines/github/demisto/content/34542/workflows/83843403-a4ba-49ea-9a65-732022b50d43/jobs/150587

**Comment by ShahafBenYakir** [comment]
Reverting changes done to illuminate pack to demonstrate the solution

File: Tests/Marketplace/search_and_install_packs.py [hunk 10]
```diff
@@ -121,13 +110,15 @@ def get_pack_dependencies(client, prints_manager, pack_data, thread_index, lock)
         lock.release()
 
 
-def search_pack(client, prints_manager, pack_display_name, thread_index, lock):
+def search_pack(client: demisto_client, prints_manager: ParallelPrintsManager, pack_display_name: str, pack_id: str,
+                thread_index: int, lock: Lock):
```
Comment (Itay4) [inline Tests/Marketplace/search_and_install_packs.py:114]:
```suggestion
                thread_index: int, lock: Lock) -> dict:
```

**Review by Itay4 — APPROVED** [review]
